r/src/Makevars.in

```diff
@@ -0,0 +1,21 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+PKG_CPPFLAGS=@cflags@
+PKG_CXXFLAGS=$(C_VISIBILITY)
+CXX_STD=CXX11
+PKG_LIBS=@libs@ -Wl,-rpath,/usr/local/lib
```

Comment:
Hard-coded rpath?

Reply:
I don't recall exactly why the issue here did you try linking without this linker option entirely?

Reply:
That works locally, I just sent a travis job to see what happens 🤞

Reply:
🤔 maybe not, getting this now locally: 

```
** testing if installed package can be loaded
Error: package or namespace load failed for ‘arrow’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Library/Frameworks/R.framework/Versions/3.5/Resources/library/arrow/libs/arrow.so':
  dlopen(/Library/Frameworks/R.framework/Versions/3.5/Resources/library/arrow/libs/arrow.so, 6): Library not loaded: @rpath/libarrow.10.dylib
  Referenced from: /Library/Frameworks/R.framework/Versions/3.5/Resources/library/arrow/libs/arrow.so
  Reason: image not found
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library/arrow’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.5/Resources/library/arrow’

Exited with status 1.
```

@jimhester is this what you had previously in rarrow perhaps ?

Reply:
What does `otool -L arrow.so` give you?

Reply:
```
romain@purrplex ~/git/apache/arrow/r/src $ otool -L arrow.so
arrow.so:
	arrow.so (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libarrow.10.dylib (compatibility version 10.0.0, current version 10.0.0)
	/Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libR.dylib (compatibility version 3.5.0, current version 3.5.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1454.90.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
```